### DISCUSSION TITLE
is there any cookbook about CDT core jar

### DISCUSSION THREAD

**Discussion by a77goddess** [eclipse-cdt/cdt#170] [Q&A] [answered]
Hi, there, is there any docs about CDT DOM architecture?
I'm working on parsing c/c++ source code with CDT core jar, but I have not found any cookbook or docs that introducing how to use classes in `org.eclipse.cdt.core.*` . 
many thanks for replying!!!

**Comment by jonahgraham** [comment, marked as answer]
Your best bet is [old presentations about CDT](https://wiki.eclipse.org/CDT:Presentations).

I can recommend these as an entry points:

- [Using CDT APIs to programmatically introspect C/C++ code](https://wiki.eclipse.org/images/c/c7/CDT_APIs_for_code_introspection.pdf) - Markus Schorn at EclipseCon 2009
- [Codan: a C/C++ Static Analysis Framework for CDT](https://docs.google.com/viewer?a=v&pid=explorer&chrome=true&srcid=1UNayw6WbckeiBLb2psf0xsyGmFf9dApwoB2BbaipU0-8c-fu3bxzCz9eSt9Q&hl=en) - Alena Laskavaia at [EclipseCon 2011](http://www.eclipsecon.org/2011/sessions/?page=sessions&id=2041)
- [Parsing and Analyzing C/C++ code in Eclipse](https://wiki.eclipse.org/images/e/ec/McMaster_2012_invited_talk_cdt.pdf) - Mike Kucera at [McMaster University](http://www.cas.mcmaster.ca/~cs4tb3/2011_12/Times.html), March 2012

There is also this FAQ entry [I am developing a plug-in that requires a parsing a c file and obtaining the information such as method signatures, parameter types etc. How can I do this using the CDT parser?](https://github.com/eclipse-cdt/cdt/blob/main/FAQ/README.md#i-am-developing-a-plug-in-that-requires-a-parsing-a-c-file-and-obtaining-the-information-such-as-method-signatures-parameter-types-etc-how-can-i-do-this-using-the-cdt-parser) - I think it is relevant, but it also may be  dated too.

**Reply by a77goddess** [reply]
owesome, the entry points above helped me a lot, very thankful!!!
